File: android/src/main/java/org/maplibre/mlrn/components/mapview/MLRNAndroidTextureMapViewManager.java
package org.maplibre.mlrn.components.mapview;

import com.facebook.react.bridge.ReactApplicationContext;
import org.maplibre.android.maps.MapLibreMapOptions;
import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.annotations.ReactProp;

/**
 * Created by hernanmateo on 12/11/18.
 */

public class MLRNAndroidTextureMapViewManager extends MLRNMapViewManager {
    public static final String LOG_TAG = "MLRNAndroidTextureMapViewManager";
    public static final String REACT_CLASS = "MLRNAndroidTextureMapView";

    public MLRNAndroidTextureMapViewManager(ReactApplicationContext context) {
        super(context);
    }

    @Override
    public String getName() {
        return REACT_CLASS;
    }

    @Override
    protected MLRNAndroidTextureMapView createViewInstance(ThemedReactContext themedReactContext) {
        MapLibreMapOptions options = new MapLibreMapOptions();
        options.textureMode(true);
        return new MLRNAndroidTextureMapView(themedReactContext, this, options);
    }
}

File: package/android/src/main/java/org/maplibre/reactnative/components/mapview/MLRNAndroidTextureMapViewManager.kt
package org.maplibre.reactnative.components.mapview

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.module.annotations.ReactModule
import com.facebook.react.uimanager.ThemedReactContext
import org.maplibre.android.maps.MapLibreMapOptions

@ReactModule(name = MLRNAndroidTextureMapViewManager.REACT_CLASS)
class MLRNAndroidTextureMapViewManager(
    context: ReactApplicationContext,
) : MLRNMapViewManager(context) {
    override fun getName(): String = REACT_CLASS

    override fun createViewInstance(themedReactContext: ThemedReactContext): MLRNAndroidTextureMapView {
        val options = MapLibreMapOptions.createFromAttributes(themedReactContext)
        options.textureMode(true)

        return MLRNAndroidTextureMapView(themedReactContext, options)
    }

    companion object {
        const val LOG_TAG: String = "MLRNAndroidTextureMapViewManager"
        const val REACT_CLASS: String = "MLRNAndroidTextureMapView"
    }
}
